### PR TITLE
Fe payment stripe

### PR DESCRIPTION
**Pull request by gursharan4312**
### What this PR does (required):
- Adds Checkout and Payment pages
- Add new `/requests/:id/pay` route to pay for the request

### Screenshots / Videos (required):
image
image

### Any information needed to test this feature (required):
- navigate to `/checkout`
- Test card number 4242 4242 4242 4242
- Expiration date 04/24
- CVC 242

### Any issues with the current functionality (optional):

- Only uses mock data as of now

### CODE REVIEW

**Comment by dproc96** [comment]
Screen Shot 2021-05-26 at 8 19 03 AM
Hitting a Typescript error when I try to load it up, can you look into this?

**Comment by gursharan4312** [comment]
Type error solved.
No error is shown in my shell. Let me know if there's any other errors.

**Comment by gursharan4312** [comment]
New Changes:
- Changes Starting and Ending time in request as user can change while checkout
- Save payment details, only pay timing as of now

**Review by dproc96 — COMMENTED** [review]
Weird, I'm now being redirected to the listings page when I try to load the checkout page. Everything still looks fine on your end?

**Comment by gursharan4312** [comment]
Yeah that's due to auth context whenever you manually redirect it'll login again and redirect to listings page. I created demo checkout button in navbar just now for checking out the checkout page.

File: client/src/pages/Checkout/Payment/Payment.tsx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    setBillingDetails({
+      email: '',
+      phone: '',
+      name: '',
+    });
+    elements?.getElement('card')?.clear();
+  };
+
+  return (
+    <>
```
Comment (bonnieli) [inline client/src/pages/Checkout/Payment/Payment.tsx]: Do we need this extra div?

**Review by bonnieli — APPROVED** [review]

**Pull request merged**